### PR TITLE
Iscsi offload changes

### PR DESCRIPTION
**Pull request by teclator**
## Problem

Bugzilla: [bsc#1231385](https://bugzilla.suse.com/show_bug.cgi?id=1231385)

In the "iSCSI initiator" dialog, the offload card was not displayed correctly. Instead of bnx2i, the dialog showed "default (TCP)". I'm not sure what would have happened if I hadn't changed this to bnx2i.

In the "iBFT" dialog, the bnx2i transport was displayed correctly.

## Solution

The selection of the offload card is relevant just in case we wanted to do some discover using that specific interface.

When a different offload card is selected it calls the iscsi_offload.sh script and select the related iscsi_iface to be used by discovery commands.

But the point is that the configuration is already done during in the initialization, so, doing it on selection doesn’t change anything at all.

What is even more important, the login and configuration most of the time is done automatically by firmware and the iscsi_offload script just creates redundant iscsi iface files… therefore, after some discussion it has been decided we could get rid of the script relying on the `iscsiadm -m fw -l` and `iscsiadm -m discovery -t` fw calls.

### Out of the scope (to improve in the future)

During the review and testing we found that there are some behavior / bugs that could be addressed but not as part of this bug:

- The iface selection should allow a multiselection.
- The iBFT tab shows only the configuration of one iface instead of the whole configuration.

## Testing

- *Tested manually*

## Screenshots

image

### CODE REVIEW

File: package/yast2-iscsi-client.changes [hunk 1]
```diff
@@ -1,3 +1,12 @@
+-------------------------------------------------------------------
+Thu Oct 31 11:46:26 UTC 2024 - Knut Anderssen <kanderssen@suse.com>
+
+- Fixes for bsc##1231385
```
Comment (ancorgs) [inline package/yast2-iscsi-client.changes:4]: Double hash, will probably not pass OBS bots.

File: src/modules/IscsiClientLib.rb [hunk 2]
```diff
@@ -252,14 +260,12 @@ def GetOffloadCard
...
-        @offload_card = new_card
-        CallConfigScript() if new_card != "default"
-      end
+    def selected_iface
+      @iface || "default"
+    end
 
-      nil
+    def iface=(iface)
+      log.info "Selecting the iface: #{iface} cur:#{@iface}"
```
Comment (ancorgs) [inline src/modules/IscsiClientLib.rb:268]: A setter that actually sets nothing? This is at least surprising.

Comment (teclator) [reply src/modules/IscsiClientLib.rb:268]: cool... :see_no_evil:

[the remark this answers is not in the document]
Comment (dgdavid) [reply src/modules/IscsiClientLib.rb:268]: Or at least change "cur" by "previous" or something more understandable. Sorry for the NP

**Review by ancorgs — APPROVED** [review]
This is certainly far from being a trivial pull request. Generally it LGTM. I hope we are not overlooking something (neither in the code or at iSCSI behavior).

**Pull request merged**